behave_django/management/commands/behave.py

```diff
@@ -1,56 +1,81 @@
 from __future__ import absolute_import
 import sys
+import os
+
+from argparse import ArgumentError
+
+from distutils.version import LooseVersion
+
+from django import VERSION as DJANGO_VERSION
+from django.conf import settings
+from django.core.management.base import (BaseCommand, OutputWrapper,
+                                         CommandParser)
+from django.core.management.utils import CommandError
+from django.test.utils import get_runner
 
 from behave.configuration import options as behave_options
 from behave.__main__ import main as behave_main
-from django.core.management.base import BaseCommand
-
 from behave_django.environment import monkey_patch_behave
-from behave_django.runner import (BehaviorDrivenTestRunner,
-                                  ExistingDatabaseTestRunner,
-                                  SimpleTestRunner)
 
 
-def add_command_arguments(parser):
+# If django<2.0 mimic the django 2.2 code for better intercompatibility.
+if LooseVersion('2.0') > LooseVersion('.'.join((str(x) for x in DJANGO_VERSION[:3]))):  # noqa: E501
+    from argparse import ArgumentParser
+
+    class CommandParser(ArgumentParser):  # noqa: F811
+        """
+        Customized ArgumentParser class to improve some error messages and
+        prevent SystemExit in several occasions, as SystemExit is unacceptable
+        when a command is called programmatically.
+        """
+        def __init__(self, **kwargs):
+            self.missing_args_message = kwargs.pop('missing_args_message', None)  # noqa: E501
+            self.called_from_command_line = kwargs.pop('called_from_command_line', None)  # noqa: E501
+            kwargs.pop('allow_abbrev', False)
+            super(CommandParser, self).__init__(**kwargs)
+
+        def parse_args(self, args=None, namespace=None):
+            # Catch missing argument for a better error message
+            if (self.missing_args_message and
+                    not (args or any(not arg.startswith('-') for arg in args))):  # noqa: E501
+                self.error(self.missing_args_message)
+            return super().parse_args(args, namespace)
```

Comment:
There must be a better way than duplicating code from the Django project.

Reply:
Issue is that if i dont do this, i break dj1.11